accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,502 @@
+packaging tools to re-write to short name on pack.
+
+Specifically:
+
+* **We'll continue to use .NETCoreApp as the TFI**. This reduces the number of
+  pieces and build logic that needs to change in .NET SDK.
+
+* **net4x and earlier will continue to use .NETFramework as the TFI**. This
+  means that net4 and net5 aren't considered compatible by default, but the
+  compatibility will continue to be handled by the AssetTargetFallback in the
+  SDK/NuGet restore.
+
+* **We'll use the existing profile concept to encode the OS selection.** It's
+  worth noting that profiles today make things smaller whereas this use makes
+  things larger. We believe this can be handled in the NuGet layer by using this
+  rule for net5 and up. While that's not ideal, it reduces the number of
+  concepts we need to add.
+
+### TFMs are a closed set
+
+One question is whether third parties can extend the TFM space (i.e. the part
+after the dash) without having to rely on changes in NuGet/MSBuild. Due to the
+expansion of short name into TFI & TFM this would be non-trivial.
+
+We may open this up in the future, but for now the consensus was that we'd
+prefer to have the specific list of names that are expanded by NuGet.
+
+### What about .NET 10?
+
+Due to the fact that we're planning to bump the major version every year, we
+have to think about what will happen with version parsing in case of two digit
+version numbers, such as `net10`. Since `net10` already has a meaning (.NET
+Framework 1.0) we need to keep it that way. To avoid suprises, we'll by default
+use dottet version numbers in project templates to push developers towards being
+explicit.
+
+| Framework      | Identifier    | Version| Comment
+|----------------|---------------|--------|----------------------------------
+| net5           | .NETCoreApp   | 5.0    | Will work, but shouldn't be used.
+| net5.0         | .NETCoreApp   | 5.0    |
+| net10          | .NETFramework | 1.0    |
+| net10.0        | .NETCoreApp   | 10.0   |
+
+### Preprocessor Symbols
+
+* TODO
+
+Today, impliicit `#if` conditions get generated based off the
+`TargetFrameworkIdentifier`, so users today have:
+
+```C#
+#if NETCOREAPP
+#elif NETFRAMEWORK
+#endif
+```
+With this proposal, the `#if NETCOREAPP` condition will still be available when
+targeting `net5`, `net6`, etc and will also be turned on for `netcoreapp2.1`,
+`netcoreapp3.1`, etc. Is that intended? Do we want a new implicit condition that
+is versionless but targets all TFMs above `net5`? i.e. `#if NET`.
+
+### Persisting pre-requisites
+
+We need to require a minimum version for the iOS/Android/Windows SDKs.
+
+* Probably like how we did WinForms/WPF
+    - WinForms/WPF uses a framework reference which by design doesn't have a
+      version number
+    - We'd need something with a version number
+
+This work is captured [in this document](https://microsoft.sharepoint.com/:w:/t/DotNetTeam/EavsPfFy7lFLh7eQrdMN8QwBl05cGLPwrSzJeT8vEu32mw?e=knNQ6W).
```

Comment:
I can't access this document, it says:

> This link is only available to internal users 

Should the document be made public? Or should the link be removed?

Reply:
Right now, the doc is a brain dump and we're working on it. As soon as we have our thoughts ordered, we'll make it public, like I did with this document.